apache/superset
# feat(plugins): Time Comparison in Table Chart

### SUMMARY
We are introducing the same instant time comparison controls from `BigNumber with Time Comparison` plugin is using into the `Table Chart`. This will allow users to quickly compare numeric data in tables to fixed time ranges or a custom time range.

Same as the previous work with `BigNumber with Time Comparison` chart, these changes are behind the experimental Feature Flag and will have no effect on current charts if the FF is OFF or if the control is not activated in the panel.

Different from the `BigNumber with Time Comparison` , we wont have multiple queries built in the frontend sent to the API, instead, we're expanding the original queries to send a new field that would indicate whether or not the query should produce a JOINed query with between the ranges.

Any new viz type we want to use these new controls, we must use single query approach, just `BigNumber with Time Comparison` should be using the multiple queries approach until a follow up PR updates it.

The main benefit of using single query and not multiple is that we can create the aforementioned JOIN statements to prepare correct results of queries, with the JOINS we can address comparison queries with Server Pagination for example which is not possible with running queries in isolation.

This PR must be followed by other non-blocking adjustments that are still in the work:
1. Make use of a control that will display the time ranges use in both queries
2. Make use of the `DATEDIFF` function to build the comparison time range
3. Enhancements to UI with expand/collapse features and others


### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
New metrics:
<img width="1699" alt="Screenshot 2024-03-01 at 14 11 51" src="https://github.com/apache/superset/assets/38889534/22c747a1-6f84-488f-89c2-59948565105d">

New Metrics with Server Pagination:
<img width="1699" alt="Screenshot 2024-03-01 at 14 13 01" src="https://github.com/apache/superset/assets/38889534/9c1c8c83-c85e-4a0f-8a80-67cdb55373a9">

Regular Table if FF is OFF or the control is not being used:
<img width="1699" alt="Screenshot 2024-03-01 at 14 14 25" src="https://github.com/apache/superset/assets/38889534/bbf6a26b-9728-40db-9cb7-7cefe186c1df">


### TESTING INSTRUCTIONS

1. Make sure the Fature Flag: `CHART_PLUGINS_EXPERIMENTAL` is ON
2. Create a table chart and use the new controls to display comparison metrics

### ADDITIONAL INFORMATION
